source/mqtt_client_connection.c

```diff
@@ -374,9 +382,11 @@ PyObject *aws_py_mqtt_client_connection_new(PyObject *self, PyObject *args) {
     Py_DECREF(self_proxy);
 proxy_new_failed:
 use_websockets_failed:
+set_on_closed_failed:
```

Comment:
I've come to hate this pattern for cleanup.  We did it early on, but it is far better (imo) to have a failure path that can safely be executed from all branch points.  If scoping is an issue, declarations can be moved up to the top of the function.

Would much rather see something like this:

```
on_failure:

   Py_XDECREF(self_proxy); 
   aws_mqtt_client_connection_release(py_connection->native);
   aws_mem_release(allocator, py_connection);
   return NULL;
```